File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example/h10-w/system/control_power.cpp
// control_power.cpp

#include <iostream>
#include <iomanip>
#include <thread>
#include <chrono>
#include "h10-w/h10w_system.h"
#include "common/common_def.h"

using namespace humanoid;
using namespace humanoid::h10w;

int main() {
  std::cout << "=== 上半身上电/下点控制及状态获取示例 ===" << std::endl;
  H10wSystem system;

  std::cout << "正在进行上半身上电控制..." << std::endl;
  if (system.powerControl(PowerState::POWER_ON) == 0) {
    std::cout << "上半身上电控制成功" << std::endl;
  } else {
    std::cerr << "上半身上电控制失败" << std::endl;
  }

  std::this_thread::sleep_for(std::chrono::milliseconds(50));

  std::cout << "正在获取上半身电源状态..." << std::endl;
  PowerState state;
  if (system.getPowerState(state) == 0) {
    std::cout << "获取上半身电源状态："
              << (state == PowerState::POWER_ON ? "上电" : "下电") << std::endl;
  } else {
    std::cerr << "获取上半身电源状态失败" << std::endl;
  }

  std::this_thread::sleep_for(std::chrono::seconds(3));

  std::cout << "正在进行上半身下电控制..." << std::endl;
  if (system.powerControl(PowerState::POWER_OFF) == 0) {
    std::cout << "上半身下电控制成功" << std::endl;
  } else {
    std::cerr << "上半身下电控制失败" << std::endl;
  }

  std::this_thread::sleep_for(std::chrono::milliseconds(50));

  std::cout << "正在获取上半身电源状态..." << std::endl;
  PowerState state2;
  if (system.getPowerState(state2) == 0) {
    std::cout << "获取上半身电源状态："
              << (state2 == PowerState::POWER_ON ? "上电" : "下电")
              << std::endl;
  } else {
    std::cerr << "获取上半身电源状态失败" << std::endl;
  }

  return 0;
}

File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example_py/h10-w/system/control_power.py
import humanoid_sdk_py
import humanoid_sdk_py.h10w as h10w
import time

h10w_system = h10w.H10wSystem()
time.sleep(1)

print("=== 上半身上电/下点控制及状态获取示例 ===")

ret = h10w_system.powerControl(humanoid_sdk_py.PowerState.POWER_ON)
if ret == 0:
    print("上半身上电控制成功")
else:
    print("上半身上电控制失败")

time.sleep(0.05)

ret, state = h10w_system.getPowerState()
if ret == 0 and state is not None:
    print(
        "获取上半身电源状态：",
        "上电" if state == humanoid_sdk_py.PowerState.POWER_ON else "下电",
    )
else:
    print("获取上半身电源状态失败")

time.sleep(1)

ret = h10w_system.powerControl(humanoid_sdk_py.PowerState.POWER_OFF)
if ret == 0:
    print("上半身上电控制成功")
else:
    print("上半身上电控制失败")

time.sleep(0.05)

ret, state = h10w_system.getPowerState()
if ret == 0 and state is not None:
    print(
        "获取上半身电源状态：",
        "上电" if state == humanoid_sdk_py.PowerState.POWER_ON else "下电",
    )
else:
    print("获取上半身电源状态失败")
